### PR TITLE
chore: bump minimum Python version from 3.10 to 3.13

### PR DESCRIPTION
**Pull request by helper2424**
Updates requires-python, classifiers, ruff target-version, mypy python_version, all CI workflow PYTHON_VERSION env vars and setup steps, and pyupgrade args to target Python 3.13.

## Title

Short, imperative summary (e.g., "fix(robots): handle None in sensor parser"). See [CONTRIBUTING.md](../CONTRIBUTING.md) for PR conventions.

## Type / Scope

- **Type**: (Bug | Feature | Docs | Performance | Test | CI | Chore)
- **Scope**: (optional — name of module or package affected)

## Summary / Motivation

- One-paragraph description of what changes and why.
- Why this change is needed and any trade-offs or design notes.

## Related issues

- Fixes / Closes: # (if any)
- Related: # (if any)

## What changed

- Short, concrete bullets of the modifications (files/behaviour).
- Short note if this introduces breaking changes and migration steps.

## How was this tested (or how to run locally)

- Tests added: list new tests or test files.
- Manual checks / dataset runs performed.
- Instructions for the reviewer

Example:

- Ran the relevant tests:

  ```bash
  pytest -q tests/ -k <keyword>
  ```

- Reproduce with a quick example or CLI (if applicable):

  ```bash
  lerobot-train --some.option=true
  ```

## Checklist (required before merge)

- [ ] Linting/formatting run (`pre-commit run -a`)
- [ ] All tests pass locally (`pytest`)
- [ ] Documentation updated
- [ ] CI is green

## Reviewer notes

- Anything the reviewer should focus on (performance, edge-cases, specific files) or general notes.
- Anyone in the community is free to review the PR.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -121,7 +121,6 @@ reachy2 = ["reachy2_sdk>=1.0.15,<1.1.0"]
 kinematics = ["lerobot[placo-dep]"]
 intelrealsense = [
     "pyrealsense2>=2.55.1.6486,<2.57.0 ; sys_platform != 'darwin'",
-    "pyrealsense2-macosx>=2.54,<2.55.0 ; sys_platform == 'darwin'",
```
Comment (helper2424) [inline pyproject.toml:124]: Here is an attempt to add 3.13 wheels for macos - https://github.com/cansik/pyrealsense2-macosx/pull/12

Comment (helper2424) [reply pyproject.toml:124]: Another option - migrate to 3.12 python

Comment (helper2424) [reply pyproject.toml:124]: Solved - the 3.13/3.14 have new versions

File: src/lerobot/envs/factory.py [hunk 2]
```diff
@@ -93,8 +93,8 @@ def make_env_pre_post_processors(
             )
         )
 
-    preprocessor = PolicyProcessorPipeline(steps=preprocessor_steps)
-    postprocessor = PolicyProcessorPipeline(steps=postprocessor_steps)
+    preprocessor = DataProcessorPipeline[dict[str, Any], dict[str, Any]](steps=preprocessor_steps)
+    postprocessor = DataProcessorPipeline[dict[str, Any], dict[str, Any]](steps=postprocessor_steps)
```
Comment (imstevenpmwork) [inline src/lerobot/envs/factory.py:97]: Any reason for this change?

File: src/lerobot/processor/pipeline.py [hunk 3]
```diff
@@ -51,10 +51,6 @@
 from .converters import batch_to_transition, create_transition, transition_to_batch
 from .core import EnvAction, EnvTransition, PolicyAction, RobotAction, RobotObservation, TransitionKey
 
-# Generic type variables for pipeline input and output.
-TInput = TypeVar("TInput")
-TOutput = TypeVar("TOutput")
-
```
Comment (imstevenpmwork) [inline src/lerobot/processor/pipeline.py:57]: Have we checked that the linter is able to resolve the type even if we remove this ?

File: src/lerobot/envs/factory.py (hunk 2 above)
Comment (helper2424) [reply src/lerobot/envs/factory.py:97]: I haven't checked all fixed carefully, some of them were made semiautomatically. I want to check all of them firstly ,a dn go through all tutorials, to be sure that it's fine.

**Comment by helper2424** [comment]
@imstevenpmwork regarding the 3.10/.3.11 questions - https://devguide.python.org/versions. I am not sure that many projects use 3.10 at the moment, also. 3.10 lifecycle will be finished this year - so definitely need to migrate.

Lerobot has versioning itself, so if u need older version of python - u can use older version of the library. I think that up to 3.12 migration won't bring any issues for anyobody. Maybe some projects still use 3.12, so here is could a bean issue.

But in any case we can discuss it.

**Comment by imstevenpmwork** [comment]
Let's do `3.12`: https://github.com/huggingface/lerobot/pull/3023
